### ISSUE TITLE
Custom widget rendering

### ISSUE THREAD

**Issue by erkin98** [datalayer/jupyter-ui#43]
Hi, is it possible to render custom widgets in react app?

**Comment by echarles** [comment]
Yes, it is possible and there are various ways to create those custom widgets - Happy to give more details on those options (or hear more about your requirements).

IPyWidgets

<img width="1026" alt="Screenshot 2023-02-02 at 17 59 40" src="https://user-images.githubusercontent.com/226720/216396627-c89de605-1411-46db-bf33-06229bae641f.png">

Panel

<img width="1039" alt="Screenshot 2023-02-02 at 17 51 44" src="https://user-images.githubusercontent.com/226720/216396646-78807ac3-8cd3-42f4-a7b8-ac8b7e15c3cd.png">


Mime renderers

<img width="1024" alt="Screenshot 2023-02-02 at 18 21 44" src="https://user-images.githubusercontent.com/226720/216396676-0a2f5c50-a34a-46d8-958d-5d1a2e879e5e.png">

Something a bit more evolved with a one page editor.

<img width="779" alt="Screenshot 2023-02-02 at 18 24 35" src="https://user-images.githubusercontent.com/226720/216397275-8be9240e-3b77-49a5-b1c7-a9c05b8b8f73.png">

**Comment by erkin98** [comment]
what if i create custom views and modules in both sides py and ts, how can i use them?

**Comment by echarles** [comment]
ipywidget and panel are the 2 existing options to achieve communication between python server and the user interface. jupyter-react allows you to use those 2 technologies. Would that work for you?

**Comment by erkin98** [comment]
ipywidgets would be greate
